cloud/client.go

```diff
@@ -169,6 +169,9 @@ func (c *client) doCall(method, url string, opts ...requestOpt) (int, string, er
 		}
 
 		if status == http.StatusUnauthorized {
+			if !r.hasAuth {
```

Comment:
Should we combine this with the below if? e.g.
```suggestion
			if !r.hasAuth || alreadyReAuthed {
```

it's not clear why `alreadyReAuthed` returns `err` and not `errUnauthorized`; I would have thought they would both return the same error.

Reply:
Will give it a try

Reply:
Added this